src/Controller/Component/AuthComponent.php

```diff
@@ -147,7 +154,8 @@ class AuthComponent extends Component
         'logoutRedirect' => null,
         'authError' => null,
         'unauthorizedRedirect' => true,
-        'storage' => 'Session'
+        'storage' => 'Session',
+        'earlyAuth' => false
```

Comment:
Would it be crazy to have an option named `authenticateIn` that takes `beforeFilter` or `startup` as values?